### PR TITLE
Feature/civ 4220 about you and claim

### PR DESCRIPTION
**Pull request by RosDiaf**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/routes/features/public/claim/claimantPartyTypeController.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      break;
+    case CounterpartyType.SOLE_TRADER:
+      res.redirect(CLAIMANT_SOLE_TRADER_DETAILS_URL);
+      break;
+    case CounterpartyType.COMPANY:
+      res.redirect(CLAIMANT_COMPANY_DETAILS_URL);
+      break;
+    case CounterpartyType.ORGANISATION:
+      res.redirect(CLAIMANT_ORGANISATION_DETAILS_URL);
+      break;
```
Comment (commitBlob) [inline src/main/routes/features/public/claim/claimantPartyTypeController.ts:23]: can you please add default case, it is probably unlikely to happen but still we want to capture it

File: src/test/unit/views/features/public/claim/claimant-party-type.test.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      expect(htmlDocument.title).toEqual(`Your money claims account - ${t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.PAGE_TITLE')}`);
+    });
+
+    it('should display header', () => {
+      const header = htmlDocument.getElementsByClassName('govuk-heading-l');
+      expect(header[0].innerHTML).toContain(t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.TITLE'));
+    });
+
+    it('should display paragraph', () => {
+      const body = htmlDocument.getElementsByClassName('govuk-body');
```
Comment (commitBlob) [inline src/test/unit/views/features/public/claim/claimant-party-type.test.ts:42]: can you please refactor line 42 into ` const body = mainWrapper.getElementsByClassName('govuk-body');`

File: src/test/unit/views/features/public/claim/claimant-party-type.test.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      expect(radioButtons[1].innerHTML).toContain(t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.CLAIMING_AS_SOLE_TRADER_OR_SELF_EMPLOYED'));
+      expect(radioButtons[1].innerHTML).toContain(t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.CLAIMING_FOR_EXAMPLE_TRADEPERSON'));
+      expect(radioButtons[2].innerHTML).toContain(t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.CLAIMING_AS_LIMITED_COMPANY'));
+      expect(radioButtons[2].innerHTML).toContain(t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.CLAIMING_FOR_EXAMPLE_COMPANY'));
+      expect(radioButtons[3].innerHTML).toContain(t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.CLAIMING_AS_ORGANISATION'));
+      expect(radioButtons[3].innerHTML).toContain(t('PAGES.CLAIMANT_PARTY_TYPE_SELECTION.CLAIMING_FOR_EXAMPLE_PARTNERSHIP_TRUST_CHARITY_CLUB_OR_ASSOCIATION'));
+    });
+
+    it('should display save and continue button', () => {
+      const button = htmlDocument.getElementsByClassName('govuk-button')[0];
```
Comment (commitBlob) [inline src/test/unit/views/features/public/claim/claimant-party-type.test.ts:60]: can you please refactor line 60 into `const button = mainWrapper.getElementsByClassName('govuk-button')[0];`

File: src/main/routes/features/public/claim/claimantPartyTypeController.ts (hunk 1 above)
Comment (RosDiaf) [reply src/main/routes/features/public/claim/claimantPartyTypeController.ts:23]: Added

File: src/test/unit/views/features/public/claim/claimant-party-type.test.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  describe('on GET', () => {
+    let htmlDocument: Document;
+    beforeEach(async () => {
+      nock(idamUrl)
+        .post('/o/token')
+        .reply(200, { id_token: citizenRoleToken });
+      const response = await request(app).get(CLAIMANT_PARTY_TYPE_SELECTION_URL);
+      const dom = new JSDOM(response.text);
+      htmlDocument = dom.window.document;
```
[the remark this answers is not in the document]
Comment (RosDiaf) [reply src/test/unit/views/features/public/claim/claimant-party-type.test.ts:30]: Added

File: src/test/unit/views/features/public/claim/claimant-party-type.test.ts (hunk 2 above)
Comment (RosDiaf) [reply src/test/unit/views/features/public/claim/claimant-party-type.test.ts]: Added

File: src/test/unit/views/features/public/claim/claimant-party-type.test.ts (hunk 3 above)
Comment (RosDiaf) [reply src/test/unit/views/features/public/claim/claimant-party-type.test.ts]: Added

File: src/main/modules/i18n/locales/cy.json [hunk 5]
```diff
@@ -457,6 +457,19 @@
       "LATEST_UPDATE": "Diweddariad diweddaraf",
       "VIEW_CLAIM": "Gweld yr hawliad"
     },
+    "CLAIMANT_PARTY_TYPE_SELECTION": {
+      "TITLE": "About you and this claim",
```
Comment (GorilaNaranja) [inline src/main/modules/i18n/locales/cy.json:461]: Please alphabetic order

File: src/main/routes/features/public/claim/claimantPartyTypeController.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    default:
+      res.render('not-found');
+      break;
+  }
+};
+
+claimantPartyTypeController.get(CLAIMANT_PARTY_TYPE_SELECTION_URL, (req: express.Request, res: express.Response) => {
+  // TODO: get from DraftStore
```
Comment (GorilaNaranja) [inline src/main/routes/features/public/claim/claimantPartyTypeController.ts:31]: Do we have claim ID to get and save it in Redis?

File: src/main/modules/i18n/locales/cy.json (hunk 5 above)
Comment (RosDiaf) [reply src/main/modules/i18n/locales/cy.json]: Changes added

**Review by GorilaNaranja — APPROVED** [review]

File: src/main/routes/features/public/claim/claimantPartyTypeController.ts (hunk 6 above)
Comment (RosDiaf) [reply src/main/routes/features/public/claim/claimantPartyTypeController.ts:31]: Once the model is created, the code will be refactored with the service.

**Pull request merged**